cpp/src/parquet/column_reader.cc

```diff
@@ -1971,9 +1975,27 @@ class TypedRecordReader : public TypedColumnReaderImpl<DType>,
       this->ConsumeBufferedValues(values_to_read);
     }
 
+    current_rg_processed_records_ += records_read;
     return records_read;
   }
 
+  int64_t ReadRecordDataWithSkipCheck(const int64_t num_records) {
```

Comment:
Originally we can safely use `ReadRecords()` and `SkipRecords()` without any assumption. Now we have mixed `ReadRecords()` into `ReadRecords()` and users can no longer call `SkipRecords()` blindly. Should we throw in the SkipRecords() if record_skipper_ has been enabled and use a custom SkipRecordsXXX here instead?

@fatemehp Do you have any concern on this?